src/hotspot/cpu/riscv/c2_MacroAssembler_riscv.cpp

```diff
@@ -3093,3 +3093,56 @@ void C2_MacroAssembler::extract_fp_v(FloatRegister dst, VectorRegister src, Basi
     vfmv_f_s(dst, tmp);
   }
 }
+
+void C2_MacroAssembler::round_double_mode_v(VectorRegister dst, BasicType dst_bt, uint vector_length,
+                                            VectorRegister src, int round_mode,
+                                            VectorRegister tmp1, VectorRegister tmp2) {
+
+  assert_different_registers(dst, src);
+  assert_different_registers(tmp1, tmp2);
+  vsetvli_helper(dst_bt, vector_length);
+
+  // The vector implementation is similar to the scalar version,
+  // check C2_MacroAssembler::round_double_mode for more details
+  // Set rounding mode for conversions
+  // Here we use similar modes to double->long and long->double conversions
+  // Different mode for long->double conversion matter only if long value was not representable as double,
+  // we got long value as a result of double->long conversion so, it is definitely representable
+  switch (round_mode) {
+    case RoundDoubleModeNode::rmode_ceil:
+      csrwi(CSR_FRM, C2_MacroAssembler::rup);
+      break;
+    case RoundDoubleModeNode::rmode_floor:
+      csrwi(CSR_FRM, C2_MacroAssembler::rdn);
+      break;
+    case RoundDoubleModeNode::rmode_rint:
+      csrwi(CSR_FRM, C2_MacroAssembler::rne);
```

Comment:
No need to set the CSR here as `FRM` has been set to Round to Nearest mode when enter Java code. And if you set `FRM` to some other rounding modes, you will need to restore it to Round to Nearest mode after processing. Check [JDK-8330094](https://bugs.openjdk.org/browse/JDK-8330094) for more details. But the problem is that modifying CSR on RISC-V is very costly.